File: app/src/main/java/com/viafourasample/src/activities/main/MainViewModel.java
package com.viafourasample.src.activities.main;

import com.viafourasdk.src.ViafouraSDK;
import com.viafourasdk.src.services.auth.VFAuthService;

public class MainViewModel {
    private VFAuthService auth = ViafouraSDK.auth();

    public void getAuthState(VFAuthService.UserLoginStatusCallback callback){
        auth.getUserLoginStatus(callback);
    }
    public void logout(){
        auth.logout();
    }
}

File: app/src/main/java/com/viafourasample/src/activities/main/MainViewModel.kt
package com.viafourasample.src.activities.main

import com.viafourasdk.src.ViafouraSDK
import com.viafourasdk.src.services.auth.VFAuthService

class MainViewModel {
    private val auth: VFAuthService = ViafouraSDK.auth()

    fun getAuthState(callback: VFAuthService.UserLoginStatusCallback) {
        auth.getUserLoginStatus(callback)
    }

    fun logout() {
        auth.logout()
    }
}
